File: scripts/apply_migration.py
#!/usr/bin/env python3
"""
应用大型 SQL 迁移文件的脚本
"""

import os
import sys

# 读取 SQL 文件
sql_file_path = os.path.join(os.path.dirname(__file__), '../supabase/migrations/20009_restore_create_tenant_schema_final.sql')

with open(sql_file_path, 'r', encoding='utf-8') as f:
    sql_content = f.read()

print(f"✅ 读取 SQL 文件成功")
print(f"📝 文件大小: {len(sql_content)} 字符")
print(f"📝 文件行数: {sql_content.count(chr(10))} 行")

# 输出 SQL 内容（用于调试）
print("\n" + "="*80)
print("SQL 内容预览:")
print("="*80)
print(sql_content[:500])
print("...")
print(sql_content[-500:])
print("="*80)

print("\n✅ SQL 文件读取完成")
print("💡 请手动通过 supabase_apply_migration 工具应用此迁移")

File: scripts/apply-migration.js
/**
 * 应用大型迁移文件的脚本
 */

const fs = require('fs');
const path = require('path');
const { createClient } = require('@supabase/supabase-js');

// 从环境变量读取 Supabase 配置
require('dotenv').config();

const supabaseUrl = process.env.TARO_APP_SUPABASE_URL;
const supabaseKey = process.env.TARO_APP_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error('❌ 错误：未找到 Supabase 配置');
  console.error('请确保 .env 文件中包含 TARO_APP_SUPABASE_URL 和 TARO_APP_SUPABASE_ANON_KEY');
  process.exit(1);
}

const supabase = createClient(supabaseUrl, supabaseKey);

async function applyMigration() {
  try {
    console.log('📖 读取迁移文件...');
    const migrationPath = path.join(__dirname, '../supabase/migrations/20009_restore_create_tenant_schema_final.sql');
    const migrationSQL = fs.readFileSync(migrationPath, 'utf8');
    
    console.log('📝 迁移文件大小:', migrationSQL.length, '字符');
    console.log('🚀 开始应用迁移...');
    
    // 执行 SQL
    const { data, error } = await supabase.rpc('exec_sql', { sql: migrationSQL });
    
    if (error) {
      console.error('❌ 迁移失败:', error);
      process.exit(1);
    }
    
    console.log('✅ 迁移成功应用！');
    console.log('📊 结果:', data);
    
  } catch (err) {
    console.error('❌ 发生错误:', err.message);
    process.exit(1);
  }
}

applyMigration();
